Source/Common/compression.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#include "pch.h"
+
+#if !HC_NOZLIB
+
+#if HC_PLATFORM == HC_PLATFORM_WIN32 || HC_PLATFORM == HC_PLATFORM_GDK
+
+#include "compression.h"
+
+// TODO Raul - Address Warning
+#pragma warning (disable: 4189)
+
+NAMESPACE_XBOX_HTTP_CLIENT_BEGIN
+
+Compression::Compression()
+{
+}
+
+void Compression::CompressToGzip(uint8_t* inData, uInt inDataSize, HCCompressionLevel compressionLevel, http_internal_vector<uint8_t>& outData)
+{
+    http_internal_vector<uint8_t> buffer;
+    uint32_t compressionLevelValue = static_cast<std::underlying_type<HCCompressionLevel>::type>(compressionLevel);
+
+    const size_t BUFSIZE = 128 * 1024;
+    uint8_t temp_buffer[BUFSIZE];
```

Comment:
It still feels like you might be doing more copying than necessary here.  Could you initialize 'buffer' to be BUFSIZE, pass a pointer directly to that and the resize as needed for subsequent calls to deflate?